pyccl/tracers.py

```diff
@@ -262,6 +321,95 @@ def get_bessel_derivative(self):
             return []
 
         return np.array([t.der_bessel for t in self._trc])
+	
+    def MG_add_tracer(self, cosmo, kernel, mg_transfer, 
+                    k_MG=None, der_bessel=0, der_angles=0,
+                    bias_transfer_a=None, bias_transfer_k=None):
+        """ function to set mg_transfer in the right format and add MG tracers 
+            for different cases including different cases and biases like intrinsic 
+            alignements (IA) when present
+        """
+        # case with no astro biases
+        if ((bias_transfer_a is None) and (bias_transfer_k is None)):    
+            if (isinstance(k_MG, (list, np.ndarray))): 		
+                # Scale-dependent case with full k array 
+                self.add_tracer(cosmo, kernel,
+                    transfer_ka=mg_transfer, der_bessel=der_bessel,
+                    der_angles=der_angles)
+            elif (isinstance(k_MG, float) or isinstance(k_MG, int)): 		
+                # Scale-dependent case with  a single specified k value
+                self.add_tracer(cosmo, kernel,
+                    transfer_a=mg_transfer, der_bessel=der_bessel,
+                    der_angles=der_angles)
+            else:		
+                # Scale-independent case where k=None
+                self.add_tracer(cosmo, kernel,
+                    transfer_a=mg_transfer, der_bessel=der_bessel, 
+                    der_angles=der_angles)
+
+        #  case of an astro bias depending on a and  k 
+        elif ((bias_transfer_a is not None) and (bias_transfer_k is not None)):   
+            if (isinstance(k_MG, (list, np.ndarray))):
+                # Scale-dependent case with full k array 
+                mg_transfer_new = (mg_transfer[0], mg_transfer[1],
+                                (bias_transfer_a[1] * (bias_transfer_k[1]*mg_transfer[2]).T).T)
+                self.add_tracer(cosmo, kernel,
+                    transfer_ka=mg_transfer_new,
+                    der_bessel=der_bessel, der_angles=der_angles)	
+            elif (isinstance(k_MG, float) or isinstance(k_MG, int)):
```

Comment:
Again, these second two cases seem to have the same outcome?

Reply:
see response above.